cpp/number_theory/gcd.cpp

```diff
@@ -0,0 +1,5 @@
+int gcd(int a, int b) { 
```

Comment:
Could you please convert this also to a template function so that either integer or long both can be accepted as input using the same function call ?

Reply:
Replaced int with template